src/fidesops/db/base_class.py

```diff
@@ -23,16 +24,16 @@
 
 def get_key_from_data(data: Dict[str, Any], cls_name: str) -> str:
     """
-    Extracts key from data and slugifies. If no key, uses a slugified name.
+    Extracts key from data, validates. If no key, uses a slugified name.
 
     Will be used as the URL slug on applicable classes.
     """
-    key = slugify(data.get("key")) if data.get("key") else None
+    key = FidesKey.validate(data.get("key")) if data.get("key") else None
```

Comment:
This highlights the importance of reaching a consensus soon on how we're going to handle versioning and releases 
https://github.com/ethyca/fides/discussions/223.

If someone had already started using fidesops, this would be a pretty big breaking change, as they might have resources created with keys that follow separate rules.

Other places affected by this change include  `quickstart.py`, the postman collections and https://github.com/ethyca/fidesdemo.

Reply:
Ah, just seeing https://github.com/ethyca/fides/discussions/223 now. Taking a look

Reply:
So https://github.com/ethyca/fidesops/pull/53 is also in the same boat- we'll need to decide on versioning before merging that in, too, right?

Reply:
An argument could be made that it's still very early and we likely don't have fidesops adopters yet (that's what we were saying about #53), so these changes can go ahead and go in.  But the sooner we get to a state where our API is stable and we're being very intentional about versioning and communicating breaking changes, the better.

Reply:
Created a new ticket to make sure the other assets are updated when this is merged https://github.com/ethyca/fidesops/issues/65

Reply:
thanks for making this ticket @eastandwestwind!